### PR TITLE
Implement List.product and Array.product

### LINKED ISSUE #362: [Stdlib] Implement List.product and Array.product
Here's the type for `List.product`:

```grain
product : (List<a>, List<b>) -> List<(a, b)>
```

`List.product` should take two lists, and return a new list of tuples of all of the combinations of elements from the two lists.

Similarly, here's the type for `Array.product`:

```grain
product : (Array<a>, Array<b>) -> Array<(a, b)>
```

### PR DESCRIPTION
**Pull request by Dony477**
Implemented product function on array and list. Closes #362

### CODE REVIEW

File: stdlib/list.gr [hunk 1]
```diff
@@ -246,4 +246,14 @@ let findIndex = (fn, list) => {
...
   findItemIndex(list, 0)
-}
\ No newline at end of file
+}
+
+let product = (a, b) => {
+  let mut list = []
+  forEach((aItem) => {
+    forEach((bItem) => {
+     list = append(list, [(aItem, bItem)])
```
Comment (ospencer) [inline stdlib/list.gr:255]: Here we could instead do `list = [(aItem, bItem), ...list]` and call `reverse` at the end for a bit more efficiency 🙂

**Review by ospencer — CHANGES_REQUESTED** [review]
Awesome work! I left one note, but other than that, this is fantastic 👏🏽

File: stdlib/list.gr (hunk 1 above)
Comment (Dony477) [reply stdlib/list.gr]: @ospencer  good point 👍

**Review by ospencer — APPROVED** [review]
Sweet, thank you!

**Comment by Dony477** [comment]
@ospencer there were some conflicts but i resolved them :)

**Review by ospencer — COMMENTED** [review]
Thanks for fixing that! 😄

**Pull request merged**